Sign in errors

cheddartot opened the pull request.

cheddartot committed:
Added errors for bad input on sign-in and login

Added errors for bad username/password, username already being taken on sign up, and passwords not matching on sign up.

cheddartot committed:
Moved functions and changed location logic

Moved a few redundant calls to utils functions, and made it so sign up doesn't add a location to the database if an identical location already exists.

pgayee694 commented on app/routes.py:
```diff
@@ -70,13 +70,16 @@ def signup():
     form_sign_up = SignInForm()
     if current_user.is_authenticated:
         return redirect('/')
+    if not(utils.find_user_by_username(form_sign_up.username.data) is None):
```
simplify this: in python you can do null checks by just doing "if object"

pgayee694 commented on app/routes.py:
```diff
@@ -70,13 +70,16 @@ def signup():
         return redirect('/')
+    if not(utils.find_user_by_username(form_sign_up.username.data) is None):
+        return render_template('signup.html', title='Sign Up', form=form_sign_up, user_exists=True)
+    if not(form_sign_up.password.data == form_sign_up.password2.data):
```
simplify

tkorisko reviewed (approved):
👌👌

cheddartot committed:
Simplified some expressions

simplified some boolean expressions because I didn't know how None worked

pgayee694 commented on app/routes.py:
```diff
@@ -70,13 +70,16 @@ def signup():
         return redirect('/')
+    if utils.find_user_by_username(form_sign_up.username.data):
+        return render_template('signup.html', title='Sign Up', form=form_sign_up, user_exists=True)
+    if form_sign_up.password.data != form_sign_up.password2.data:
```
put a line of space between different if blocks

pgayee694 commented on app/routes.py:
```diff
@@ -57,10 +57,10 @@ def login():
         if user is None or not user.check_password(form_log_in.password.data):
             flash('invalid username or password')
-            return redirect('/login/')
+            return render_template('login.html', title='Log In', form=form_log_in, invalid=True)
```
you should display error messages to the user like this via flash(). Here's a good example: https://stackoverflow.com/questions/40949746/how-to-display-flashing-message-without-reloading-the-page-in-flask

cheddartot committed:
Changed messages to use flash()

Changed messages to use flash(), and removed unecessary flash() calls. This should also solve an issue in the Friends branch where that page was showing multiple flash messages.

cheddartot committed:
Merge branch 'master' into Sign-in-Errors
